src/main/java/bio/overture/songsearch/graphql/EntityDataFetcher.java

```diff
@@ -49,15 +55,35 @@ public DataFetcher getDataFetcher() {
                   }
                   if (RUN_ENTITY.equals(values.get("__typename"))) {
                     final Object runId = values.get("runId");
+                    val ids = getRelevantAnalysisIdsFromRunParameters(values.get("parameters"));
+
                     if (runId instanceof String) {
                       return new Run(
                           (String) runId,
                           analysisService.getAnalysesByRunId((String) runId),
-                          "Not Implemented!");
+                          analysisService.getAnalysesById(ids)
+                          );
                     }
                   }
                   return null;
                 })
             .collect(toList());
   }
+
+  @SuppressWarnings("unchecked")
```

Comment:
while the likelihood of this not being a Map<String, Obj> is close to zero, still we should check and handle it properly otherwise it will throw uncaught class cast which is no bueno